test/metabase/db/schema_migrations_test.clj

```diff
@@ -2429,3 +2429,37 @@
                          (is (t2/exists? (t2/table-name :model/Field) :id defective-field-id)))
                        (testing "Migrating up again should still work"
                          (migrate!))))))))))))))
+
+(deftest deactivate-defective-duplicates-test
+  (testing "Migration v49.2024-06-27T00:00:09"
+    (impl/test-migrations ["v49.2024-06-27T00:00:09"] [migrate!]
+      (let [db-id         (t2/insert-returning-pk! :metabase_database
+                                                   {:details    "{}"
+                                                    :created_at :%now
+                                                    :updated_at :%now
+                                                    :engine     "h2"
+                                                    :is_sample  false
+                                                    :name       "some_db"})
+            table         (t2/insert-returning-instance! :metabase_table
+                                                         {:db_id      db-id
+                                                          :name       "some_table"
+                                                          :created_at :%now
+                                                          :updated_at :%now
+                                                          :active     true})
+            field!        (fn [values]
+                            (t2/insert-returning-instance! :metabase_field
+                                                           (merge {:table_id      (:id table)
+                                                                   :active        true
+                                                                   :parent_id     nil
+                                                                   :base_type     "type/Text"
+                                                                   :database_type "TEXT"
+                                                                   :created_at    :%now
+                                                                   :updated_at    :%now}
+                                                                  values)))
+            active+field [[true  (field! {:name "x",  :is_defective_duplicate true, :nfc_path "[\"x\",\"y\""})]
+                          [false (field! {:name "x", :is_defective_duplicate true, :nfc_path nil})]
```

Comment:
Could you add a field where `is_defective_duplicate = false` to check that it's not going to updated?

Reply:
yep, 405c5aa72b3b065e25a569ff6d5e034c89a3c638